### ISSUE TITLE
cython 0.29 produces invalid __path__, breaking nose

### ISSUE THREAD

**Issue by jbaum-cmcrc** [cython/cython#2662]
Cython 0.29 produces invalid `__path__` module attributes, breaking nosetests (and possibly other tools).

According to the Python reference, "`__path__` must be an iterable of strings, but it may be empty" ([Python language reference, section 5.4.5](https://docs.python.org/3.6/reference/import.html#module-path)).

* In cython 0.29, the `__path__` attribute on a compiled module is `None`, which is not an iterable.
* In cython 0.28.5, there is no `__path__` attribute on a compiled module.

Expected behaviour: Compiled modules have a valid `__path__` attribute (or omit it altogether).

Impact on nosetests:
* When collecting tests to run, nose checks the `__path__` attribute on each module; if it exists, nose attempts to iterate over it in order to find submodules to test. (If the attribute doesn't exist, nose skips the iteration.)
* In cython 0.29, nose attempts to iterate the invalid `__path__` attribute, giving a "'NoneType' object is not iterable" error.
* In cython 0.28.5, there is no `__path__` attribute on a compiled module, so nose skips the iteration and proceeds without error; the new behaviour is therefore a regression as far as projects using nosetests are concerned.
* Error message given by `nose` when testing a module compiled by Cython 0.29:
  ```
    File ".../lib/python3.6/site-packages/nose/loader.py", line 343, in loadTestsFromModule
      for module_path in module_paths:
  TypeError: 'NoneType' object is not iterable
  ```

**Comment by hmaarrfk** [comment]
It seems that this line of code
https://github.com/cython/cython/blob/c9eda12b4806fba3931f7b791be7a672c92d9ca0/Cython/Compiler/Symtab.py#L1169

was recently added and affects `__path__`

**Comment by jbaum-cmcrc** [comment]
That seems like it should be an easy fix to initialise this one to `()` instead of `None`, which would make it valid (and acceptable to nose), though not informative.

**Comment by hmaarrfk** [comment]
Yeah, but i would like to hear something from a Cython developer. They will probably want a test to ensure it doesn't keep breaking things in the future.

**Comment by robertwb** [comment]
Yes, we should be setting this to something valid, if anything at all. (It should probably typically be omitted.) And, yes, we'd want tests with any fix :).

**Comment by scoder** [comment]
Could also just be this code: https://github.com/cython/cython/blob/da657c8e326a419cde8ae6ea91be9661b9622504/Cython/Utility/ModuleSetupCode.c#L975

According to PEP 302, only packages (incl. namespace packages) are supposed too have a `__path__`. We should therefore avoid setting it at all for normal modules.

**Comment by scoder** [comment]
Hopefully fixed in 792263575b900eafd99ee0bbfbff1f14fcd70cc3.
